services/payments/statemachine_zebpay.go

```diff
@@ -62,7 +65,17 @@ func (zm *ZebpayMachine) Pay(ctx context.Context) (*paymentLib.AuthenticatedPaym
 		case zebpay.TransferPendingCode:
 			// Set backoff without changing status
 			zm.backoffFactor = zm.backoffFactor * 2
+		case zebpay.TransferNotFoundCode:
+			// The transaction was never received by zebpay. We should not have gotten into this
```

Comment:
this comment seems inaccurate. is it not the case that we call `CheckTransfer` before `submit`?